connectors/src/resources/base_resource.ts

```diff
@@ -56,4 +64,6 @@ export abstract class BaseResource<M extends Model> {
       },
     });
   }
+
+  abstract toJSON(): unknown;
```

Comment:
We could actually enforce the type a bit more with
```suggestion
  abstract toJSON(): Record<string, unknown>;
```

Reply:
:+1: